### PR TITLE
Core: additional properties and generic updater

### PR DESCRIPTION
**Pull request by yugangw-msft**
3 things
1. For `additional properties`: if it is empty, do not show; otherwise flatten out to the parent scope to reflect the real json payload on the wire. This not only avoids the confusion reported by compute team, but also avoids the unnecessary breaking change from CLI's end if the SDK gets regenerated from newer swagger which authors related new field. 
2. Adjust generic update to work with it. I had to manually install older SDK to have new properties show up as additional and test it works
3. Fix 2 existing bugs in the generic update which I have marked out in the PR
    

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by promptws** [comment]
View a preview at https://prompt.ws/r/Azure/azure-cli/6277
This is an **experimental preview** for @microsoft users.

File: src/azure-cli-core/azure/cli/core/commands/arm.py [hunk 1]
```diff
@@ -744,12 +747,13 @@ def remove_properties(instance, argument_values):
...
     if not list_index:
-        _find_property(instance, list_attribute_path)
+        property_val = _find_property(instance, list_attribute_path)
         parent_to_remove_from = _find_property(instance, list_attribute_path[:-1])
         if isinstance(parent_to_remove_from, dict):
             del parent_to_remove_from[list_attribute_path[-1]]
         elif hasattr(parent_to_remove_from, make_snake_case(list_attribute_path[-1])):
-            setattr(parent_to_remove_from, make_snake_case(list_attribute_path[-1]), None)
+            setattr(parent_to_remove_from, make_snake_case(list_attribute_path[-1]),
+                    [] if isinstance(property_val, list) else None)
```
[the remark this answers is not in the document]
Comment (tjprescott) [reply src/azure-cli-core/azure/cli/core/commands/arm.py]: Previously you should be able to with
```
az vm update -g yg -n yg-2 --set storageProfile.dataDisks="[]"
```

File: src/azure-cli-core/azure/cli/core/tests/test_generic_update.py [hunk 2]
```diff
@@ -63,6 +63,11 @@ def __init__(self):
         self.empty_dict_of_dicts = {'dict': {'dict2': None}}
         self.empty_dict = {'dict3': None}
 
+        self.additional_properties = {'additional1': 'addition value #1'}
+
+    def enable_additional_properties_sending(self):
+        pass
```
Comment (tjprescott) [inline src/azure-cli-core/azure/cli/core/tests/test_generic_update.py]: Is this intentionally a no-op?

Comment (yugangw-msft) [reply src/azure-cli-core/azure/cli/core/tests/test_generic_update.py]: Okay, let me just make our test object be an instance of SDK's Model type. This not only gets rid of this confusion, but also mirrors the most common update scenarios

**Pull request merged**